src/s3/s3_object.rs

```diff
@@ -26,6 +26,14 @@ impl S3Object {
             key: key.as_ref().trim_start_matches('/').to_owned(),
         }
     }
+
+    pub fn bucket(&self) -> &str {
+        &self.bucket
+    }
+
+    pub fn key(&self) -> &str {
+        &self.key
+    }
```

Comment:
**question:** What's the motivation for the additional methods here? My preference would be to keep the API as small as possible and have the user access the fields directly for this relatively simple struct.

```suggestion
```

Reply:
Exposing values through getters rather than directly accessing public fields ensures better encapsulation. However, removing the `pub` fields would break backward compatibility, so we must retain them.